cadc-storage-adapter-test/src/main/java/org/opencadc/inventory/storage/test/StorageAdapterBasicTest.java

```diff
@@ -445,4 +447,160 @@ public void testIteratorBucketPrefix() {
             Assert.fail("unexpected exception: " + ex);
         }
     }
+    
+    /**
+     * Sub-classes of this test can override this method and add @Test to
+     * test delete and recover.
+     */
+    protected void testDeleteRecover() {
+        long datalen = 8192L;
+        DateFormat df = DateUtil.getDateFormat(DateUtil.IVOA_DATE_FORMAT, DateUtil.UTC);
+        try {
+            URI artifactURI = URI.create(TEST_NAMESPACE + "TEST/testDeleteRecover");
+            NewArtifact na = new NewArtifact(artifactURI);
+            na.contentLength = datalen;
+            StorageMetadata sm = adapter.put(na, TestUtil.getInputStreamOfRandomBytes(datalen), null);
+            Assert.assertFalse(sm.deletePreserved);
+            
+            
+            Thread.sleep(2000L);
+            adapter.delete(sm.getStorageLocation());
+            
+            try {
```

Comment:
I don't see the point of the try/catch block... if the exception is expected, why is no action performed when the exception does not occur?